Cargo features (#11867)
mesonbuild/meson: docs/markdown/Wrap-dependency-system-manual.md

dependency_names = foo-bar-rs
```

> Cargo features are exposed as Meson boolean options, with `feature-` prefix. For

Reviewer: ```suggestion
Cargo features are exposed as Meson boolean options, with the `feature-` prefix. For
```

Author: done

> example the `default` feature is named `feature-default` and can be set from
> command line with `-Dfoo-rs:feature-default=false`. When a cargo subproject

Reviewer: ```suggestion
example the `default` feature is named `feature-default` and can be set from the
command line with `-Dfoo-rs:feature-default=false`. When a cargo subproject
```

depends on another cargo subproject, it will automatically enable features it

> needs using `dependency('foo-rs', default_options: ...)` mechanism. However,

Reviewer: ```suggestion
needs using the `dependency('foo-rs', default_options: ...)` mechanism. However,
```

unlike Cargo, the set of enabled features is not managed globally. Let's assume
the main project depends on `foo-rs` and `bar-rs`, and they both depend on

> `common-rs`. The main project will first lookup for `foo-rs` which itself will

Reviewer: ```suggestion
`common-rs`. The main project will first look up `foo-rs` which itself will
```

Author: done

> configure `common-rs` with a set of features. Later, when `bar-rs` lookups for
> `common-rs` it has already been configured and the set of features cannot be

Reviewer: ```suggestion
configure `common-rs` with a set of features. Later, when `bar-rs` does a lookup for
`common-rs` it has already been configured and the set of features cannot be
```

Author: done

changed. If `bar-rs` wants extra features from `common-rs`, Meson will error out.
It is currently the responsability of the main project to resolve those
issues by enabling extra features on each subproject:
```meson
project(...,
  default_options: {
    'common-rs:feature-something': true,
  },
)
```

> In addition, if `meson/meson.build` file exists, Meson will call `subdir('meson')`

Reviewer: ```suggestion
In addition, if the file `meson/meson.build` exists, Meson will call `subdir('meson')`
```

Author: done

where the project can add manual logic that would usually be part of `build.rs`.

> Some naming conventions needs to be respected:

Reviewer: ```suggestion
Some naming conventions need to be respected:
```

Author: done

- `extra_args` variable is pre-defined and can be used to add any Rust arguments.
  This is typically used as `extra_args += ['--cfg', 'foo']`.
- `extra_deps` variable is pre-defined and can be used to add extra dependencies.

>   This is typically used as `extra_deps += dependency('foo')`.

Reviewer: "the `xxx` variable is pre-defined"

Author: done
